src/cachedev.rs

```
@@ -97,19 +97,40 @@ impl FromStr for CacheTargetParams {
         let origin_dev = parse_device(vals[3], "origin sub-device for cache target")?;
 
         let block_size = Sectors(parse_value(vals[4], "data block size")?);
-        let num_feature_args: usize = parse_value(vals[5], "number of feature args")?;
+
+        let num_feature_args = if vals.len() == 5 {
+            0
+        } else {
+            parse_value::<usize>(vals[5], "number of feature args")?
+        };
 
         let end_feature_args_index = 6 + num_feature_args;
-        let feature_args: Vec<String> = vals[6..end_feature_args_index]
-            .iter()
-            .map(|x| (*x).to_string())
-            .collect();
+
+        if vals.len() <= end_feature_args_index + 1 {
```

Just make this ```if vals.len() < end_feature_args_index``` and then make the error message specific to too few feature_args. That change means we could still fail on policy missing, but I want to leave that until later, also.

---

To avoid an out-of-bounds panic on test inputs that are only incorrect because of an incorrect feature argument number, this condition was chosen. Changing the condition to ```if vals.len() < end_feature_args_index``` would make sense for test inputs that are incorrect for the additional reason of being void of mandatory policy args, e.g. "cache 42:42 42:43 42:44 16 4 writethrough".